internal/hook/item_hook_handler.go

```diff
@@ -482,11 +478,15 @@ func GroupRestoreExecHooks(
 	}
 	hookFromAnnotation := getPodExecRestoreHookFromAnnotations(metadata.GetAnnotations(), log)
 	if hookFromAnnotation != nil {
-		byContainer[hookFromAnnotation.Container] = []NamedExecRestoreHook{
+		// default to first container in pod if unset
+		if hookFromAnnotation.Container == "" {
```

Comment:
edge case where this can segfault it `pod.Spec.Containers` is empty.

Reply:
L472 above ensures there is at least 1 container in the pod spec.